include/Vector.h

```diff
@@ -0,0 +1,59 @@
+#pragma once
+#include <compare>
+#include <iostream>
+
+class Vector
+{
+
+public:
+	// ========== c-tors & d-tor ===============
+	explicit Vector(size_t size = 0, int value = 0);
+	Vector(const Vector& other);
+	Vector(Vector&& other) noexcept;
+	~Vector();
+
+	// ========== methods ======================
+	int* data();
+	const int* data() const;
+	bool empty() const;
+	size_t size() const;
+	size_t capacity() const;
+	void clear();
+	void reserve(size_t size);
+	void resize(size_t size, int value = 0);
+	void swap(Vector& other) noexcept;
+	void push_back(int value);
+	void pop_back();
+	void insert(size_t pos, int value);
+	void erase(size_t pos);
+
+	// ========= operators overloading =========
+	Vector& operator=(const Vector& other);
+	Vector& operator=(Vector&& other) noexcept;
+
+	int& operator[](size_t index);
```

Comment:
You are correct that the operator[] signature returns T&, but specifically for a trivially-copyable type like int there is no need to return a reference, the compiler will just omit it and copy the value

Reply:
fixed it, but such line will no longer work: 
v2[1] = 5;